Sources/TracingInstrumentation/Span.swift

```diff
@@ -190,9 +192,11 @@ public enum SpanAttribute: Equatable {
     case array([SpanAttribute])
     case stringConvertible(CustomStringConvertible)
 
+    #if compiler(>=5.2)
     /// This is a "magic value" that is used to enable the KeyPath based accessors to specific attributes.
     /// This value will never be stored or returned, and any attempt of doing so would WILL crash your application.
     case __namespace
+    #endif
```

Comment:
not sure we should do this about the enum case; keep it in, even if we dont use it.

Also, we should see if we can remove this __namespace -- it's a bit of a hack.
Can you make a ticket for it? "See if we can remove the case __namespace in attributes"?

Reply:
https://github.com/slashmo/gsoc-swift-tracing/issues/129

Reply:
I left this out on purpose because it's not being used anywhere but the 5.2 "magic". What would be the benefit of leaving it in for < 5.2?

Reply:
If you have a library that supports all swift versions, and they happen to switch over the attributes enum, then they'd be unable to write ONE switch implementation that works on all compiler versions without using `default` since they have to handle `__namespace` on 5.2+ and they couldn't handle it on previous versions because it's compiled out.

We should keep enums always the same across all supported compiler versions